app/Http/Controllers/TeamController.php

```diff
@@ -42,6 +44,38 @@ public function create()
         ]);
     }
 
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return Response
+     */
+    public function joinForm()
+    {
+        return Inertia::render('team/Join');
+    }
+
+    /**
+     * Background Route for joining Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function join(Request $request)
+    {
+        $data = $this->validate($request, [
+            'code' => ['required', 'string', 'exists:teams,join_code'],
+        ]);
+
+        $team = Team::where('join_code', $data['code'])->first();
+
+        $user = $request->user();
+
+        if(!$user->teams->contains($team->id)){
+            $user->teams()->attach([$team->id]);
+        }
```

Comment:
[`syncWithoutDetaching`](https://laravel.com/docs/8.x/eloquent-relationships#syncing-associations) is exactly there for this case :)

```suggestion
        $user->teams()->syncWithoutDetaching([$team->id]);
```